File: AÑO 2/programacion_y_diseño_orientado_a_objetos/Practicas/Napakalaki_Java/src/NapakalakiGame/Monster.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package NapakalakiGame;

/**
 *
 * @author guillermogotre
 */
public class Monster {
    private String name;
    private int combatLevel;
    private Prize prize;
    private BadConsequence badConsequence;
    private int levelChangeAgainstCultistPlayer = 0;
    
    public Monster(String name, int level, BadConsequence bc, Prize prize){
        this.name = name;
        this.combatLevel = level;
        this.badConsequence = bc;
        this.prize = prize;
    }
    
    public Monster(String n,int l, BadConsequence badConsequence, Prize p, int IC){
        this.name = n;
        this.combatLevel = l;
        this.badConsequence = badConsequence;
        this.prize = p;
        this.levelChangeAgainstCultistPlayer = IC;
    }
    
    public int getCombatLevelAgainstCultistPlayer(){
        return getCombatLevel() + levelChangeAgainstCultistPlayer;
    }

    public String getName() {
        return name;
    }

    public int getCombatLevel() {
        return combatLevel;
    }

    public Prize getPrize() {
        return prize;
    }

    public BadConsequence getBadConsequence() {
        return badConsequence;
    }
    
    public int getLevelsGained(){
        return prize.getLevel();
    }
    
    public int getTreasuresGained(){
        return prize.getTreasures();
    }

    @Override
    public String toString() {
        return "Monster{" + "name=" + name + ", combatLevel=" + (combatLevel + this.levelChangeAgainstCultistPlayer) + ", prize=" + prize + ", badConsequence=" + badConsequence + '}';
    }
    
}

File: AÑO 2/programacion_y_diseño_orientado_a_objetos/Practicas/Napakalaki_Ruby/pkg/Napakalaki-0.0.1/lib/monster.rb
#encoding: utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

class Monster
  attr_reader :name, :combatLevel, :prize, :badConsequence
  def initialize(name,combatLevel,bc,prize)
    @name=name
    @combatLevel=combatLevel
    @prize=prize
    @badConsequence=bc
  end
  
  def getLevelsGained
    return @prize.level
  end
  
  def getTreasuresGained
    return @prize.treasures
  end
  
  def to_s
    "Monster{name=#{@name}, combatLevel=#{@combatLevel}, prize=#{@prize.to_s}, badConsequence=#{@badConsequence.to_s}}";
  end
end
